### ISSUE TITLE
--dataflowServiceOptions=use_runner_v2 is broken

### ISSUE THREAD

**Issue by damccorm** [apache/beam#21700]
Per [https://lists.apache.org/list.html?user@beam.apache.org](https://lists.apache.org/list.html?user@beam.apache.org) passing `–dataflowServiceOptions=use_runner_v2` alone triggers pipeline errors. This is surprising as it should be an alias for experiments, and furthermore no action should be required on the SDK side to produce a runner v2 compatible graph.

Imported from Jira [BEAM-14421](https://issues.apache.org/jira/browse/BEAM-14421). Original Jira may contain additional context.
Reported by: robertwb.

**Comment by kennknowles** [comment]
@robertwb this is fixed, I hope?

**Comment by TheNeuralBit** [comment]
CC: @y1chi who posted some context in the jira

**Comment by y1chi** [comment]
I think the fix is either turn on  the same set of flags of --experiments=use_runner_v2 when seeing --dataflowServiceOptions=use_runner_v2 in SDK so we get proper FNAPI job types or do it on service side. Do it on service has slight advantage of working for released SDKs.

**Comment by kennknowles** [comment]
Yes, the expectation of `dataflowServiceOptions` is that the SDK will _never_ do anything based on them. This ensures that everything is handled in the service and can be updated for released SDKs.

**Comment by kennknowles** [comment]
@robertwb any update?

**Comment by robertwb** [comment]
No updates on this. Despite being GA I think we can continue to recommend experiments for this flag until it is fully rolled out by default. I'm on the fence whether it's a good idea to patch the SDK to fix this for future SDKs only.

**Comment by kennknowles** [comment]
Perhaps this should not be a longstanding P1 if it is not something we intend to change?

**Comment by robertwb** [comment]
Long term, we don't care as we plan to change the default.

**Comment by kennknowles** [comment]
I interpret that to mean that setting to P2 and unassigning reflects our intentions.
